Makefile

```diff
@@ -19,11 +19,33 @@ type: check-pre-commit
 format:
 	@find src test -name '*.py' | xargs autopep8 --in-place
 
+OSBUILD_GEN_DEPSOLVE_DNF4_EXECUTABLE := $(shell (which "osbuild-gen-depsolve-dnf4" 2>/dev/null))
+OSBUILD_MAKE_DEPSOLVE_DNF4_RPM_STAGE_EXECUTABLE := $(shell (which "osbuild-make-depsolve-dnf4-rpm-stage" 2>/dev/null))
+OSBUILD_MAKE_DEPSOLVE_DNF4_CURL_SOURCE_EXECUTABLE := $(shell (which "osbuild-make-depsolve-dnf4-curl-source" 2>/dev/null))
+
+.PHONY: check_binaries
+check_binaries:
```

Comment:
I understand what it's doing and why it's only for the Python ones as the Go ones are already a dependency of the `test` target but I'd like it more to have `check_binaries` be a dependency of `external` (which is called by `test`).

Reply:
No reason - I just started to understand what's happening here...
actually I would want to have it build it straight forward, if missing. How `make` should work anyhow
I just changed it from "building" it to just an error message as I wasn't sure if I should force people to use the local `venv`

But if you are fine with it I'll directly implement

`venv` missing -> `python3 -m venv venv`
-> always `. venv/bin/activate` in the makefile
`external/*` missing -> `pip install …`

Reply:
It's an OK approach though you can drop the activation in the Makefile and use explicit paths instead.